when possible, surface volume handle as provider id for PVs

ameijer opened the pull request:
## What does this PR change?
* a request by by the azure team for the resource ID to be surfaced via the API 


## How will this PR impact users?
* ProviderID will no longer always be the name. In cases where the volume handle is available, that will be used as the provider id 


## How was this PR tested?
* 

## Does this PR require changes to documentation?
* 

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

Commit:
disabled healthz endpoint logging

Signed-off-by: Andrii Chubatiuk <andrew.chubatiuk@motional.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

klaasjand committed:
feat: add custom s3 export endpoint

Signed-off-by: Klaas Jan Dijksterhuis <klaasjand@users.noreply.github.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

klaasjand committed:
feat: add custom s3 export endpoint

Signed-off-by: Klaas Jan Dijksterhuis <klaasjand@users.noreply.github.com>

nickcurie committed:
add ip field to load balancers

Signed-off-by: nickcurie <nick.curie64@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

nickcurie committed:
add ip field to load balancers

Signed-off-by: nickcurie <nick.curie64@gmail.com>

nikovacevic committed:
GTM-52 Fix parsing of GCP pricing and create new node diagnostic pricing API

Signed-off-by: Niko Kovacevic <nikovacevic@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
when possible, surface volume handle as provider id for PVs

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
add to query

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

avrodrigues5 committed:
fix subquery fallback to not have dependency on metric relabel if users doesnt provide it in BYO prometheus (#2076)

Signed-off-by: Alan Rodrigues <alanr5691@yahoo.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
add filtering

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

nikovacevic commented on pkg/costmodel/cluster.go:
```diff
@@ -1436,10 +1441,6 @@ func pvCosts(diskMap map[DiskIdentifier]*Disk, resolution time.Duration, resActi
 		}
 
 		diskMap[key].Cost = cost * (diskMap[key].Bytes / 1024 / 1024 / 1024) * (diskMap[key].Minutes / 60)
-		providerID, _ := result.GetString("provider_id") // just put the providerID set up here, it's the simplest query.
-		if providerID != "" {
-			diskMap[key].ProviderID = provider.ParsePVID(providerID)
-		}
```
I think that this is going to break reconciliation. Sorry for taking so long on this review... I wanted to find the proof first.

nikovacevic replied:
Here's my reasoning: Today, reconciliation of disks works by fetching CUR data, including so-called "provider ID" and cost, and matching that data with Asset data on this field. If today the match works, and I have no reason to believe it doesn't, then we cannot change the value here without also changing it in reconciliation. (Or else, this change works by virtue of the fact that it does not change the value of provider ID, in which case it is moot.)

Does that make sense? We could equivalently try to run this, and see if the provider ID values change -- and if they do, then we can check if reconciliation breaks.

nikovacevic replied:
You might check with @Sean-Holcomb on cloud IDs and which fields are and are not available in the CUR, too. He may know more than I do here.

avrodrigues5 committed:
fix non deletion of nodes from node metrics due to dependency added on node arch types

Signed-off-by: Alan Rodrigues <alanr5691@yahoo.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

avrodrigues5 committed:
fix non deletion of nodes from node metrics due to dependency added on node arch types

Signed-off-by: Alan Rodrigues <alanr5691@yahoo.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

klaasjand committed:
feat: changed custom s3 endpoint naming and scheme to alts3, used aws sdk based s3 client endpoint construction

Signed-off-by: Klaas Jan Dijksterhuis <klaasjand@users.noreply.github.com>

klaasjand committed:
feat: changed custom s3 endpoint naming and scheme to alts3, used aws sdk based s3 client endpoint construction

Signed-off-by: Klaas Jan Dijksterhuis <klaasjand@users.noreply.github.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

klaasjand committed:
refactor: renamed path arguments to filePath in order to prevent import alias for path

Signed-off-by: Klaas Jan Dijksterhuis <klaasjand@users.noreply.github.com>

klaasjand committed:
refactor: renamed path arguments to filePath in order to prevent import alias for path

Signed-off-by: Klaas Jan Dijksterhuis <klaasjand@users.noreply.github.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

alrs committed:
pkg/cloud/alibaba: fix dropped errors

Signed-off-by: Lars Lehtonen <lars.lehtonen@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

nikovacevic committed:
GTM-63 sanitize label names before metric emission to prevent label name collisions

Signed-off-by: Niko Kovacevic <nikovacevic@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

saweber committed:
improve auth error handling and response for AWS orphaned resources

Signed-off-by: saweber <saweber@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

mbolt35 committed:
GCP Provider Cluster Filter Fix (#2116)

Unsure as to why this went untested, but this should correctly fix the query formatting.

Signed-off-by: Matt Bolt <mbolt35@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
Added special case for unallocated/unassigned aliased fields when filtering. Added additional test case for filtering into an unallocated aliased field. (#2114)

Signed-off-by: Ashleigh <ashleigh@kubecost.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

nikovacevic committed:
BURNDOWN-209 fix Inf and negative values in pod-to-PVC attribution

Signed-off-by: Niko Kovacevic <nikovacevic@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

ameijer committed:
add quarterly accum opt (#2120)

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

thomasvn committed:
When unable to properly remove a metric from the CostModelMetricsEmitter, log an Error. (#2109)

Signed-off-by: thomasvn <thomasnguyen96@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

thomasvn committed:
When unable to properly remove a metric from the CostModelMetricsEmitter, log an Error. (#2109)

Signed-off-by: thomasvn <thomasnguyen96@gmail.com>
Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
revert asset based surfacing of provider ID

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
corrected checkout

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
re-write to pass through allocations, not assets

Signed-off-by: Alex Meijer <ameijer@kubecost.com>

Commit:
bingen work

ameijer commented:
closing in favor of https://github.com/opencost/opencost/pull/2126
